test/functional/test_framework/socks5.py

```diff
@@ -40,6 +41,16 @@ def __init__(self):
         self.af = socket.AF_INET # Bind address family
         self.unauth = False  # Support unauthenticated
         self.auth = False  # Support authentication
+        # An array of objects like:
+        # {
+        #     "listen_addr": "127.0.0.1"
+        #     "listen_port": 28276
+        #     "node": python_p2p_node,
+        #     "requested_to_addr": "the_client_asked_to_connect_to_this_addr.onion",
```

Comment:
I don't see `.node` used below and I don't think you really use `.requested_to_addr` anywhere after setting it below? Could this just be an array of `CAddress` or something like that?

Reply:
Both `.node` and `.requested_to_addr` members are used by the test - `.node` is used to check/query what messages `bitcoind` has sent to this peer and `.requested_to_addr` is used to check that the connection was made to an `.onion` address (and not e.g. an IPv4 address).